src/dmd/backend/cgelem.d

```diff
@@ -1680,6 +1680,82 @@ private elem *elor(elem *e, goal_t goal)
         }
     }
 
+    /* Recognize the following function and replace it with OPbswap:
+        ushort byteswap(ushort x) { return cast(ushort)(((x >> 8) & 0xFF) | ((x << 8) & 0xFF00)); }
```

Comment:
What if someone writes this instead? 

```d
ushort byteswap(ushort x) { return cast(ushort)((x >> 8) | ((x << 8) & 0xFF00)); }
```

Reply:
In fact it seems even the other mask isn't needed because the result is cast to 16 bit. So:

```d
ushort byteswap(ushort x) { return cast(ushort)((x >> 8) | (x << 8)); }
```

Reply:
There are several ways to write it. People seem endlessly creative about it. There are these choices:

1. recognize every pattern I can think of
2. recognize one pattern and encourage people to use `byteswap` whenever doing this
3. make `byteswap` an intrinsic

I figure on doing 2 and 3.